servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererObserver.java

```diff
@@ -88,17 +88,31 @@
         int ttl();
 
         /**
-         * Number of resolved records that became {@link ServiceDiscovererEvent#isAvailable() available}.
+         * Number of resolved records that became {@link ServiceDiscovererEvent.Status#AVAILABLE available}.
          *
-         * @return the number of resolved records that became {@link ServiceDiscovererEvent#isAvailable() available}
+         * @return the number of resolved records that became {@link ServiceDiscovererEvent.Status#AVAILABLE available}
          */
         int nAvailable();
 
         /**
-         * Number of resolved records that became {@link ServiceDiscovererEvent#isAvailable() unavailable}.
+         * Number of resolved records that became {@link ServiceDiscovererEvent.Status#UNAVAILABLE unavailable}.
          *
-         * @return the number of resolved records that became {@link ServiceDiscovererEvent#isAvailable() unavailable}
+         * @return the number of resolved records that became
+         * {@link ServiceDiscovererEvent.Status#UNAVAILABLE unavailable}
+         * @deprecated Implement and use {@link #nMissing()} method.
          */
-        int nUnavailable();
+        @Deprecated
+        default int nUnavailable() {
+            throw new UnsupportedOperationException("Please implement and use the nMissing() method.");
```

Comment:
This will actually break all uses of old API, because `DefaultResolutionResult` only implements `nMissing()`. We should swap `default` implementations in this interface: default `nUnavailable()` should delegate to `nMissing()`, while `nMissing()` should throw `UnsupportedOperationException`. It will be safe for our implementation, and will provide smooth migration if anyone else implements this interface.